DynamicData/Kernel/TwoStageRWLock.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace DynamicData.Kernel
```

Comment:
Although it does not show in this editor, can you remove tabs. This is easily achieved in vs. Select all text, Edit->Advanced->Untabbify Selected Lines. Please do the same for all the files which you have changed.

NB: I have never set up fx cop rules because I find it so intrusive and slows down editing.